OpenRA.Mods.Common/Traits/SpawnActorsOnSell.cs

```diff
@@ -72,6 +76,33 @@ void Emit(Actor self)
 			}
 
 			var eligibleLocations = buildingInfo.Tiles(self.Location).ToList();
+
+			var guaranteedActorTypes = Info.GuaranteedActorTypes?.Select(a =>
```

Comment:
Optional, Only do this when there are location.

Reply:
Also only do this when `dudesValue > 0`.

Reply:
the idea is to ignore dudesValue. 

And yeah we could add a few returns for optimisation. It's just that selling isn't very common so I'm not sure it'rs worth it